src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.SecretsManager.Commands.EnableAccessSecretsManager.Interfaces;
+
+namespace Bit.Core.SecretsManager.Commands.EnableAccessSecretsManager;
+
+public class EnableAccessSecretsManagerCommand : IEnableAccessSecretsManagerCommand
+{
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+
+    public EnableAccessSecretsManagerCommand(IOrganizationUserRepository organizationUserRepository)
+    {
+        _organizationUserRepository = organizationUserRepository;
+    }
+
+    public async Task<List<(OrganizationUser, string)>> EnableUsersAsync(Guid organizationId,
+        ICollection<OrganizationUser> organizationUsers)
+    {
+        var filteredUsersByOrg = organizationUsers.Where(ou => ou.OrganizationId == organizationId)
+            .ToList();
+
+        if (!filteredUsersByOrg.Any())
+        {
+            throw new BadRequestException("Users invalid.");
+        }
+
+        var filteredUsers = filteredUsersByOrg.Where(ou => ou.AccessSecretsManager == false).ToList();
```

Comment:
This is where we would usually set the `string error` property for that orgUser, e.g. "User already has access to Secrets Manager", and include it in the return value.

Then that error message displays in the bulk actions dialog on the front-end, next to that user. (You can see this in other bulk operations today, e.g. bulk confirm users who have already been confirmed.)

Reply:
Resolved in https://github.com/bitwarden/server/pull/3020/commits/015c68aa31ccdeb0b246f7d3a903eb1312174b3d